molecularnodes/mda/base.py

```diff
@@ -0,0 +1,105 @@
+"""
+Base class for MDAnalysis visualization
+"""
+
+import bpy
+import MDAnalysis as mda
+from MDAnalysis.core.groups import AtomGroup
+from ..session import get_session
+from . import utils
+from .universes import Universe, Universes
+
+
+class MDAVis:
+    """
+    Base Class for MDAnalysis visualization
+
+    Attributes
+    ----------
+    universes: Universes
+        Subscriptable and iterable list of Universe objects
+    session: MNSession
+        MolecularNodes session object
+    """
+
+    def __new__(cls) -> None:
+        """This is a singleton class"""
+        # try to retrieve object from MNSession
+        try:
+            session = get_session()
+        except AttributeError:
+            from .. import register
+
+            register()
+            session = get_session()
+        if hasattr(session, "MDAVis"):
+            return session.MDAVis
+        instance = super().__new__(cls)
+        session.MDAVis = instance  # add instance to MNSession
+        instance._initialized = False
+        instance.session = session  # link to MNSession (for deletes)
+        return instance
+
+    def __init__(self) -> None:
+        if self._initialized:
+            return
+        self._initialized = True
+        self._universes = {}  # dict to hold Universe objects created
+        self.universes = Universes(self._universes)
+        utils.subscribe_to_active_object()  # subscribe to active object changes
+
+    def add_universe(
```

Comment:
If you add the same universe twice, you will get two separate MN universes but they are both linked back to the same mda universe? Is that something we want?